cmd/webhookcert/main.go

```diff
@@ -0,0 +1,178 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"os"
+	"path"
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/config"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+)
+
+var (
+	log                            = logger.Log
+	validatingWebhookConfiguration = "nsx-operator-validating-webhook-configuration"
+)
+
+func main() {
+	log.Info("Generating webhook certificates...")
+	if err := generateWebhookCerts(); err != nil {
+		panic(err)
+	}
+}
+
+// WriteFile writes data in the file at the given path
+func writeFile(filepath string, cert *bytes.Buffer) error {
+	f, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	_, err = f.Write(cert.Bytes())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func generateWebhookCerts() error {
+	var caPEM, serverCertPEM, serverKeyPEM *bytes.Buffer
+	// CA config
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(2020),
```

Comment:
Is it safe to use the fixed number? I saw https://github.com/vmware-tanzu/nsx-operator/blob/3a7fd4e31f402c60c1062b18f40fe50ac127f1c5/pkg/nsx/auth/jwt/vcclient.go#L266 used the rand function to generate a big number (e.g. 196794602592025086508469506258673667833)..

Reply:
Thanks, done.